src/Bridge/Symfony/Routing/CachedRouteNameResolver.php

```diff
@@ -43,33 +44,8 @@ public function getRouteName(string $resourceClass, $operationType /**, array $c
         $context = \func_num_args() > 2 ? func_get_arg(2) : [];
         $cacheKey = self::CACHE_KEY_PREFIX.md5(serialize([$resourceClass, $operationType, $context['subresource_resources'] ?? null]));
 
-        if (isset($this->localCache[$cacheKey])) {
-            return $this->localCache[$cacheKey];
-        }
-
-        try {
-            $cacheItem = $this->cacheItemPool->getItem($cacheKey);
-
-            if ($cacheItem->isHit()) {
-                return $this->localCache[$cacheKey] = $cacheItem->get();
-            }
-        } catch (CacheException $e) {
-            //do nothing
-        }
-
-        $routeName = $this->decorated->getRouteName($resourceClass, $operationType, $context);
-
-        if (!isset($cacheItem)) {
-            return $this->localCache[$cacheKey] = $routeName;
-        }
-
-        try {
-            $cacheItem->set($routeName);
-            $this->cacheItemPool->save($cacheItem);
-        } catch (CacheException $e) {
-            // do nothing
-        }
-
-        return $this->localCache[$cacheKey] = $routeName;
+        return $this->getOrSave($cacheKey, $this->cacheItemPool, function () use ($resourceClass, $operationType, $context) {
```

Comment:
Perhaps `getCached`? I also agree with @soyuka that we shouldn't need to pass the `$cacheItemPool` here, it should be in the trait.